File: macroslib/tests/expectations/foreign_class_as_return_type_simple.rs
// without result Type and without "foreign" args

foreigner_class!(class Foo {
    self_type Foo;
    constructor Foo::new(_: i32) -> Foo;
    fn Foo::f(&self, _: i32, _: i32) -> i32;
});

foreigner_class!(class Boo {
    self_type Boo;

    constructor Boo::new(_: i32, _: usize) -> Boo;
    fn Boo::factory_method() -> Boo;
    fn Boo::get_one_foo(&self) -> Foo;
});

File: macroslib/tests/expectations/foreign_class_as_return_type_simple.java
r#"public static @NonNull Boo factory_method() {
        long ret = do_factory_method();
        Boo convRet = new Boo(InternalPointerMarker.RAW_PTR, ret);

        return convRet;
    }
    private static native long do_factory_method();"#;
"public final @NonNull Foo get_one_foo() {";
"public final int f(int a0, int a1)";
